File: maybe_usefulness/Money.hs
import Data.Map (Map)
import qualified Data.Map as Map
import Data.Maybe
import Data.Monoid

money = Map.fromList [("dave", Sum 10), ("gavin", Sum 11)]

{-
    {
        "dave": 10,
        "gavin": 11
    }

-}

getMoney name = Map.lookup name money

davesMoney = getMoney "dave"
gavinsMoney = getMoney "gavin"
zorbulonsMoney = getMoney "zorbulon"

total = davesMoney `mappend` gavinsMoney `mappend` zorbulonsMoney
-- => Just (Sum {getSum = 21})

\x -> x + 1

File: maybe_usefulness/Money.js
var money = {
    'dave': 10,
    'gavin': 11
};

var davesMoney = money['dave'];         // 10
var gavinsMoney = money['gavin'];       // 11
var zorbulonsMoney = money['zorbulon']; // undefined

var total = davesMoney + gavinsMoney + zorbulonsMoney;

console.log(total);
// => NaN

depositIntoBankAccount(total) // you just deposited NaN, weirdo
